tests/tap_tester/test_all_fields.py

```diff
@@ -3,19 +3,65 @@
 import tap_tester.menagerie as menagerie
 from base import TestPendoBase
 
+
 class PendoAllFieldsTest(TestPendoBase):
     def name(self):
         return "pendo_all_fields_test"
-    
+
     def test_run(self):
+        self.run_test({'accounts', 'events', 'page_events', 'pages', 'poll_events', 'track_events', 'track_types'})
+        self.run_test({"visitors", "visitor_history"}, start_date="2022-06-20T00:00:00Z")
+
+        # All these streams have similar implementation like pages and page_events so removing this test to limit the execution time
+        # self.run_test({'feature_events', 'features', 'guide_events', 'guides'})
+
+    def expected_metadata(self):
+        visitor_history = {
+            # Add back when visitor_history stream causing this test to take 4+ hours is solved,
+            # tracked in this JIRA: https://stitchdata.atlassian.net/browse/SRCE-4755
+            # Improvised the execution
+            #   - Added filtering visitors based on last updated which will reduce the execution time
+            #   - Testing visitors streams separately with latest start time
+            "visitor_history": {
+                self.PRIMARY_KEYS: {'visitor_id'},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {'modified_ts'}
+            }
+        }
```

Comment:
It's a bit confusing to remove this from the expectations in base.py then add back here. I think a more consistent pattern with what we have used in the past is to make the expected_metadata in base.py a constant source of truth for defining the expectations of a stream. Then if a stream cannot be included in a specific test, jut removing it from the streams under test.